### PR TITLE
Command channel output 

### PR DESCRIPTION
**Pull request by mappum**
This PR lets commands return channels, which allows for streaming object output. Now the add command will show hashes of files as they are added. :smile:

It's pretty satisfying to watch it add the whole go-ipfs directory.

Thanks to the sharness tests for catching various edge-case bugs I wouldn't have noticed. :+1:

### CODE REVIEW

**Comment by mappum** [comment]
Jenkins is yelling at me because one of the intermediate commits doesn't build, do you want me to change the history to fix that?

File: commands/http/handler.go [hunk 1]
```diff
@@ -97,5 +101,67 @@ func (i Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	writer.WriteString(channelHeader + ": 1\r\n\r\n")
+
+	buf := make([]byte, 32*1024)
+
+	for {
+		n, err := out.Read(buf)
+
+		if n > 0 {
+			length := fmt.Sprintf("%x\r\n", n)
+			writer.WriteString(length)
```
Comment (mappum) [inline commands/http/handler.go]: Oh, that's HTTP chunked encoding, there's a hex length before each chunk.

**Comment by jbenet** [comment]
Sweet, i'm going to merge this! Thanks! :+1:
